### ISSUE TITLE
transcr.: Treatment of specific elements

### ISSUE THREAD

**Issue by BOberreither** [karl-kraus/wpn-static#32]
Ongoing list:
- [x] `choice` > always show choice/sic, omit choice/corr
- [x] `add[not(parent::subst)]` > pipe character replaces element in text flow; text node is placed according to ```@rend```
- [x] `add[parent::subst]`  > text node is placed according to `@rend` (in add, and if there is none in subst), preceded by pipe character
- [x] general del-rule: don't include leading or trailing spaces in the crossing-out
- [x] `del[not(parent::subst)]` > text node is crossed out; pipe character is placed according to `@rend`, followed by deleatur character
- [x] `del[parent::subst]` > text node is crossed out, no extra pipe character is placed
- [x] `subst[@rend="overwritten"]` > del text node is only 50% darkness, add is placed exactly over it; font according to `@change` - currently displayed wrongly, [see e.g. page 59a](https://karl-kraus.github.io/wpn-static/idPb0059_a.html)
- [x] `restore/add` > text node is crossed out; pipe character displayed with dotted underline
- [x] `restore/del` > text node is crossed out, gets dotted underline (currently not crossed out, see [21](https://karl-kraus.github.io/wpn-static/idPb0021.html).)
- [x] `restore/subst/add+del` > del is crossed out and gets dotted underline, add-span is crossed out (for example see [page 144](https://karl-kraus.github.io/wpn-static/idPb0144.html))
- [x] ```seg[@type="transposition"and not(@subtype="implicit") and not(parent::restore)]``` > text segment highlighted by thin margin
- [x] ```seg[@type="transposition" and not(@subtype="implicit") and parent::restore]``` > text segment highlighted by dotted margin
- [x] `listTranspose` and all child nodes - omitted
- [x] `metamark`
    - [x] `@function="progress"` shows editing progress with small black or grey (according to change attr) cross placed according to `@rend` or else `@place`
    - [x] `@function="transposition"` shows mark for transposed text block; place mirrored S (`U+01A7`) according to `@rend` [for now, better symbol wanted]
    - [x] `@function="relocation"`
        - [x] `@change="edACE"` > don't display the metamark, for all others:
        - [x] pipe symbol placed inline + if `@rend`: textnode, else pipe, placed accordingly
    - [x] `@function="printInstruction"` 
        - [x] `@rendition="#typescriptFont" or "#inkFont"` - underline (see, e.g., [59a](https://karl-kraus.github.io/wpn-static/idPb0059_a.html))
        - [x] if `@spanTo` - draw vertical line from the point specified in @rend down to the height of the anchor element. Give feedback if this results in a lot of trouble.
        - [x] place it according to `@rend` or `@place`, otherwise block element (see, e.g., [59a](https://karl-kraus.github.io/wpn-static/idPb0059_a.html))
        - [x] rotate text node according to `@style`
    - [x] `@function="insertion"`
        - [x] placement inline (creates no extra spaces) and according to `@rend`
        - [x] if `@rend`: inline represented by pipe symbol (for now), at `@rend`-location: **textnode**, else pipe symbol
- [x] seg
    - [x] `@type="relocation"`
        - [ ] `@subtype="implicit"` - no further treatment  
        - [x] `@rend="line"` > add pipe symbol before text node
        - [x] `@rend="arrow"` > add arrow pointing left before text node
        - [x] `@rend="border"` > add border around text node
    - [x] `@type="transposition"` > add border around text node (color according to `@change`
- [x] 'mod'
    - [x] `@style="letterSpacing"` - underline text instead of spacing, color according to `@change`; (see e.g. [page 161](https://karl-kraus.github.io/wpn-static/idPb0161.html))
    - [x] `@style="noLetterSpacing"` - underline text instead of un-spacing; color according to `@change`; (see e.g. [page 222](https://karl-kraus.github.io/wpn-static/idPb0222.html))
    - [x] `@style='indent2'` - indent to the right - black arrow from the current end of the line to the 'target end'
    - [x] `@style='noUnderline'` - underline (nested `hi[@style='underline']`) changed to crossed-out underline (maybe as an underline + wavy underline?)
    - [x] `@rendition="#longQuote"` - don't display it as longquote (extra text block), but inline, just add a border and place a pipe according to `@rend`.
    - [x] `@rendition="#runningText1"` - don't display it as indented new paragraph, but as a "["-character placed inline at the beginning of the element and another one according to `@rend`. Don't add a border.

- [x] fw
    - [x] `@rendition="typescriptInsertNum"` - underline
- [x] hi
    - [x] `@style="underline"` - underline
- [x] gap: display U+2592 (medium shade) symbols, as many as indicated in `@extent`.
- [x] unclear: display text (if in del: strike through), but with light grey background and question mark for mouseover
- [x] `c[@resp="#edACE"]` - omit.

- [ ] mod
General: there's a metamark to be placed according to `@rend` and styled according to `@change`, and it refers to the mod via `@corresp`. They should be highlighted together. Test case exists on p 33.
    - [x] `@style="italic"` - don't display text in italics, but underline text; font size stays the same
    - [x] `@style="letterSpacing"` - underline, font size stays the same
    - [x] `@style="noLetterSpacing"` - underline, font size stays the same
    - [x] `@style="indent2"` - as is.
    - [x] `@style="noIndent"` - don't directly apply (see p. 124), but instead apply indentation from l-element. Add an arrow before the beginning of the text and another arrow placed according to `@rend`

**Comment by linxOD** [comment]
> text segment highlighted by thin margin
 > text segment highlighted by dotted margin
 
 could you explain how a thin or dotted margin should look like?
 Currently, there will be a round border circling the text with solid or dotted line.

**Comment by linxOD** [comment]
> shows mark for transposed text block;

what mark should be shown?

**Comment by BOberreither** [comment]
@linxOD Of course there is no such thing as a dotted margin, thanks for translating my gibberish to english!
For your other comment: I meant a symbol like this, "place mirrored S (U+01A7) according to @rend [for now, better symbol wanted]" - as you already did, I can see.
Also: added first specification for mod, more to follow.

**Comment by BOberreither** [comment]
@linxOD about the del-handling: currently it repeats the deleted text in the margin (or wherever it's placed), should show a [deleatur character](https://codepoints.net/U+20B0)

**Comment by linxOD** [comment]
> restore/del > text node is crossed out, gets dotted underline (doesn't yet work on [page 144](https://karlkraus.github.io/wpn-static/idPb0144.html), for example. )

this document holds restore/subst/del and not restore/del elements.
Please add handling for this cases

**Comment by BOberreither** [comment]
@linxOD About the deleatur character: in our html, it looks different than [this one](https://codepoints.net/U+20B0); is it because of the font? should not look like this: 
![Bildschirmfoto 2024-10-31 um 11 15 12](https://github.com/user-attachments/assets/42e5d72b-7acf-496e-93e8-1a8910d9079a)
but like this (just not in red):
![Bildschirmfoto 2024-10-31 um 11 15 48](https://github.com/user-attachments/assets/cebd3dde-c70f-4678-8ebf-c551b39355ef)

**Comment by linxOD** [comment]
> metamark@function='relocation':  pipe symbol placed inline + if @rend: textnode, else pipe, placed accordingly

not sure what to interpret the "+ if" part?
If there is a @rend attribute it's either in left or right margin but where does it go without? Another inline pipe at text position?

Currently added: pipe symbol for all metamark@printInstruction=relocation inline except if change=#edACE. The latter is not rendered at all.
If there is a @rend the text will be rendered in left or right margin.

**Comment by BOberreither** [comment]
> Currently added: pipe symbol for all metamark@printInstruction=relocation inline except if change=#edACE. The latter is not rendered at all. If there is a rend the text will be rendered in left or right margin.

Yes, I was unclear: If there is a `rend`-attribute, put the pipe inline and another one according to @rend. If there is a text node, replace the `@rend`-pipe with it.

**Comment by linxOD** [comment]
> Yes, I was unclear: If there is a rend-attribute, put the pipe inline and another one according to @rend. If there is a text node, replace the @rend-pipe with it.

Consider defining a unique symbol for the different cases in left and right margin but also top and bottom. This will help with the documentation and users to understand the differences.

**Comment by linxOD** [comment]
> @linxOD About the deleatur character: in our html, it looks different than [this one](https://codepoints.net/U+20B0); is it because of the font? should not look like this: ![Bildschirmfoto 2024-10-31 um 11 15 12](https://private-user-images.githubusercontent.com/37076522/381910976-42e5d72b-7acf-496e-93e8-1a8910d9079a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4ODIxNDQsIm5iZiI6MTczMDg4MTg0NCwicGF0aCI6Ii8zNzA3NjUyMi8zODE5MTA5NzYtNDJlNWQ3MmItN2FjZi00OTZlLTkzZTgtMWE4OTEwZDkwNzlhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA2VDA4MzA0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg3Y2E2MmVlZWI3MzRjYzg1NDJmNTcxYWJlMTU5MWQ4MTU2MWU3ZjE3MGNlNDViYzlhNTU5NDc4N2IzZTE5NTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.bMa-t-H5cSo9hMP1tWS2lsWjwgjl83W_qWv-vQU5gdA) but like this (just not in red): ![Bildschirmfoto 2024-10-31 um 11 15 48](https://private-user-images.githubusercontent.com/37076522/381911046-cebd3dde-c70f-4678-8ebf-c551b39355ef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4ODIxNDQsIm5iZiI6MTczMDg4MTg0NCwicGF0aCI6Ii8zNzA3NjUyMi8zODE5MTEwNDYtY2ViZDNkZGUtYzcwZi00Njc4LThlYmYtYzU1MWIzOTM1NWVmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA2VDA4MzA0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg3Mzg0ZDFlZmMwYmI2ZGRlYzBkMTU3Mjk2ZDE3N2RlMzkwMzYzNTk3MzAxMzkwZmMyYzcyY2Q1MDI0NTc4MDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.MRnrW9eam2A6PGq6IuSoeEtus8CVn2AvH9_AJNarOrU)

I guess this is a browser issue. I only see the second one. Tested in all chrome browsers and firefox.

**Comment by BOberreither** [comment]
> I guess this is a browser issue. I only see the second one. Tested in all chrome browsers and firefox.
Well, is there anything you can do to fix it to a font showing the correct symbol, even in safari?

**Comment by BOberreither** [comment]
@linxOD  I'm not going to add anything in the list above anymore, for sake of findability, but add new stuff / bugs in comments.

- [ ] See, e.g. https://karl-kraus.github.io/wpn-static/idPb0004.html, right margin, "doch", "Knechte", |i": when hovering over the element in the margin, the highlight in the text should not be between two characters, but on/around the del-character.
- [ ] metamark @function="printInstruction" with @spanTo: no "{" character in the margin; the line should not appear on hover, but always be there.
- [ ] Another treatment for longQuoteVerseCenter is necessary: Please don't center align, but just add margin left that comes as close as possible to what it looks like on page six.
- [ ] On [page 263](https://karl-kraus.github.io/wpn-static/idPb0263.html), there seems to be something wrong with the margins.

**Comment by linxOD** [comment]
> metamark function="printInstruction" with spanTo 

in case the anchor with the spanTo value is on another page. Additional handling must be implemented.

> I can add some left-indent like we do with other longQuoteVerse... elements. However, if "longQuoteVerseCenter" is reused on other pages and elments they will all start like in page six. Is this what you want?

> Page 263

The additional indent of some lg elements comes from the `mod` wrapper which also hols a rendition attribute with the value "longQuoteVerse" that includes a `seg` > `lg` holding another rendition attribute value "longQuoteVerseStart" for example. There the indent of 30px is added twice. All others are not wrapped in `mod`. 

Can you do something on the data site?

**Comment by BOberreither** [comment]
> > general del-rule: don't include leading or trailing spaces in the crossing-out
> 
> this will only work if del elements hold text only. If there are any unknown nested elementes it will apply templates for these elements.

1 out of 6 dels include other elements, that's an error rate - if these errors even occur - we can at least check out, I think. Please give it a shot!

**Comment by BOberreither** [comment]
> in case the anchor with the spanTo value is on another page. Additional handling must be implemented.

Yes, it will be - there will simply be metamark/anchor pairs for each page, so no metamarks spanning more than one side.

**Comment by BOberreither** [comment]
> I can add some left-indent like we do with other longQuoteVerse... elements. However, if "longQuoteVerseCenter" is reused on other pages and elments they will all start like in page six. Is this what you want?

Yes, please. The indent for longQuoteVerse... is like the one from paragraph first lines; the one from longQuoteVerseCenter is about double of that.

**Comment by BOberreither** [comment]
> The additional indent of some lg elements [...] Can you do something on the data site?

But why is it the same issue with verses starting from "Man läßt ihr Toben, wütend Hausen,", "Die Goldespforfen sind verrammelt,", and "So sei die Zeit in Fröhlichkeit vertan!" ?

**Comment by linxOD** [comment]
> > I guess this is a browser issue. I only see the second one. Tested in all chrome browsers and firefox.
> > Well, is there anything you can do to fix it to a font showing the correct symbol, even in safari?

The delete symbole seems to be problem on your computer. I tested the website on my apple tablet using Safari and the symbols looks very similar to version 2 as it does on chrome, edge and firefox.

Safari on Apple Tablet:
![karl-kraus-del](https://github.com/user-attachments/assets/5c2ddfba-7bd9-4dca-83fb-c750a1a0d021)

Chrome:
![Screenshot from 2024-11-22 10-23-54](https://github.com/user-attachments/assets/ea963b28-a08b-489b-9ad2-f09f9dfcc766)

**Comment by linxOD** [comment]
> > The additional indent of some lg elements [...] Can you do something on the data site?
> 
> But why is it the same issue with verses starting from "Man läßt ihr Toben, wütend Hausen,", "Die Goldespforfen sind verrammelt,", and "So sei die Zeit in Fröhlichkeit vertan!" ?

@BOberreither longQuoteVerse uses margin left and longQuoteVerseStart|End|Middle had 30px indent. I removed the indet to visualize how this looks. Since some elements have a wrapper with longQuoteVerse and additional longQuoteVerseStart or End or Middle the margin to the left increase. The question is: Should longQuoteVerseStart|End|Middle have indent? If yes, then the wrapper for longQuoteVerse must be removed. If not, longQuoteVerse must be added as wrapper.

**Comment by BOberreither** [comment]
> The question is: Should longQuoteVerseStart|End|Middle have indent? If yes, then the wrapper for longQuoteVerse must be removed. If not, longQuoteVerse must be added as wrapper.

@linxOD I think there is nothing that can be removed data-side; can you add a mechanism that prevents the additional 30px indent if the element is inside one with margin left?

**Comment by linxOD** [comment]
> mod
> General: there's a metamark to be placed according to @rend and styled according to @change, and it refers to the mod via > @corresp. They should be highlighted together. Test case exists on p 33.

As far as I could find there are 3 metamark elements with corresp but they do not refer to a mod element.

**Comment by BOberreither** [comment]
(Upcoming issues with topics discussed here will be in new issues, for the sake of legibility)
